ojmicroline_thermostat/requests/update_group_request.py

```diff
@@ -1,94 +0,0 @@
-"""GroupRequest model for OJ Microline Thermostat."""
-from __future__ import annotations
-
-from dataclasses import dataclass
-from datetime import datetime, timedelta
-from typing import Any
-
-from ..const import (
-    COMFORT_DURATION,
-    DATETIME_FORMAT,
-    REGULATION_BOOST,
-    REGULATION_COMFORT,
-    REGULATION_MANUAL,
-    REGULATION_SCHEDULE,
-    REGULATION_VACATION,
-)
-from ..models import Thermostat
-
-
-@dataclass
-class UpdateGroupRequest:
-    """UpdateGroupRequest instance."""
-
-    resource: Thermostat
-    api_key: str
-
-    def __init__(self, resource: Thermostat, api_key: str):
-        """
-        Initialise a new UpdateGroupRequest instance based on the given arguments.
-
-        Args:
-            resource: The thermostat to update.
-            api_key: The API key to use in the request.
-        """
-
-        self.resource = resource
-        self.api_key = api_key
-
-    def update_regulation_mode(
-        self,
-        regulation_mode: int,
-        temperature: int | None = None,
-        duration: int = COMFORT_DURATION,
-    ) -> dict[str, Any]:
-        """
-        Return a request body to update the regulation mode for a group.
-
-        Args:
-            regulation_mode: The mode to set the regulation mode in.
-            temperature: The temperature (for comfort and manual mode) or None.
-            duration: The duration in minutes to set the temperature
-                      for (comfort mode only), defaults to 4 hours.
-
-        Returns:
-            A JSON object containing the update.
-        """
-        if temperature is not None:
-            self.resource.temperatures[regulation_mode] = temperature
```

Comment:
Important! Previously, calling `set_regulation_mode` would actually **change** the `temperatures` variable on that thermostat.

This seemed funky to me for a couple reasons: first, this change takes effect even if the subsequent request fails due to auth failure, network unavailability, etc., and it is never rolled back in that case. Second, it feels like the `Thermostat` object should be a forever-immutable snapshot of a thermostat's state at some point in time instead of a mutable object that changes.

So… I simply removed this from the refactored code.